ipm_library/ipm_library/ipm.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2022 Hamburg Bit-Bots
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from ipm_library import utils
+from ipm_library.exceptions import NoIntersectionError
+from ipm_msgs.msg import PlaneStamped
+import numpy as np
```

Comment:
@Flova:

> Is this import formatting standard in ROS2? I find it kind of weird. Is there some guideline for this? In most python projects, they are grouped by from and import.

Reply:
Ah yes, the import orders were juggled around in https://github.com/Flova/ipm/commit/e38d324180e750b7777e80f58976e12b54ad17e6 to make flake8 happy but I don't really understand why this makes flake8 happy.

From memory, [flake8: import statements are in the wrong order](https://stackoverflow.com/questions/39739118/flake8-import-statements-are-in-the-wrong-order) was the exact issue that arose and the suggested fix in that link says to add a .flake8 file to configure flake8-import-order plugin is needed.

I couldn't get this working, but juggling around the import orders did fix the issue. (And I agree that it looks pretty bad and would like to retain the original order). Would you know much about the fix suggested on StackOverflow......?

Reply:
Created an issue so we can assign the task: https://github.com/ros-sports/ipm/issues/6